### PR TITLE
OH2-136 | Associate permission to api and check user permissions for each call

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-136.

### CODE REVIEW

**Review by mwithi — CHANGES_REQUESTED** [review]
As described in the issue, the check should be on permissions (rather than roles) and other changes are requested.

File: src/main/java/org/isf/security/UserDetailsServiceImpl.java [hunk 1]
```diff
@@ -44,6 +46,9 @@ public class UserDetailsServiceImpl implements UserDetailsService {
     @Autowired
     protected UserBrowsingManager manager;
 
+    @Autowired
+	protected PermissionManager permissionManager;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/security/UserDetailsServiceImpl.java:50]: this indentation looks incorrect.

File: src/main/java/org/isf/security/UserDetailsServiceImpl.java [hunk 2]
```diff
@@ -57,7 +62,10 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
             throw new UsernameNotFoundException(username + " was not found");
         }
         List<SimpleGrantedAuthority> authorities = new ArrayList<>();
-        authorities.add(new SimpleGrantedAuthority(user.getUserGroupName().getCode()));
+        List<Permission> permissions = permissionManager.retrievePermissionsByUsername(username);
+        for (Permission p : permissions) {
+        	  authorities.add(new SimpleGrantedAuthority(p.getName()));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/security/UserDetailsServiceImpl.java:67]: this seems to be indented too far in

Comment (ArnaudFonzam) [reply src/main/java/org/isf/security/UserDetailsServiceImpl.java]: please on my side i don't have this issue
error1

File: src/main/java/org/isf/config/SecurityConfig.java [hunk 3]
```diff
@@ -22,7 +22,10 @@
 package org.isf.config;
 
 import java.util.Arrays;
+import java.util.List;
```
Comment (mwithi) [inline src/main/java/org/isf/config/SecurityConfig.java:25]: This doesn't seem to be used, please organize imports

**Review by mwithi — APPROVED** [review]

**Review by mwithi — CHANGES_REQUESTED** [review]
just some imports to be removed

**Review by mwithi — APPROVED** [review]

**Pull request merged**